src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [hasLoadedOnce, setHasLoadedOnce] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(data => {
+        const updatedPeople = data.map(person => ({
+          ...person,
+          mother: data.find(p => p.name === person.motherName),
+          father: data.find(p => p.name === person.fatherName),
```

Comment:
The data.map() operation here is potentially very costly in terms of performance. For each person, you are searching the entire data array twice - once for the mother and once for the father. This results in a quadratic time complexity. Consider creating a lookup object first, where the keys are the names and the values are the person objects. This would allow you to find the parents in constant time.